chat.js

```diff
@@ -735,6 +756,10 @@ class CommandContext {
 				return false;
 			}
 
+			if (!this.checkFlooding(room, user) && !user.can('mute', null, room)) {
```

Comment:
why not do the ``user.can('mute', null, room)`` check first (for all of the chat moderation settings in general once at the start) and then do the different checks after if the user can't mute?

That should prevent the server from doing useless checks to only realize that the user is exempt from moderation

Reply:
Copied the one for slowchat order.

``if (!this.checkSlowchat(room, user) && !user.can('mute', null, room)) {``

Honestly, maybe it should just check the ``mute`` permission in the actual function itself. That might simplify things.

Reply:
I moved all of the permission checking in the actual function itself